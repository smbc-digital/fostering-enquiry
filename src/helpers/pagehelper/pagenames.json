[
	{
		"number": 1,
		"route": "/fostering-enquiry/about-yourself"
	},
	{
		"number": 2,
		"route": "/fostering-enquiry/contact-details"
	},
	{
		"number": 3,
		"route": "/fostering-enquiry/submit"
	}
	,
	{
		"number": 4,
		"route": "/fostering-enquiry/success"
	}
	,
	{
		"number": 5,
		"route": "/fostering-enquiry/error"
	}
]