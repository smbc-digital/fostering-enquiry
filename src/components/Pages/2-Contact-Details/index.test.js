import { React, mount} from '../../../helpers/SetupTest'
import { ContactDetails } from './index'
import { getPageRoute } from '../../../helpers/pagehelper'
import renderer from 'react-test-renderer'

describe('ContactDetails', () => {
    it('should call push on submit', () => {
        //Arrange
        const data = {
            emailAddress: {
                value:'Value',
                isValid: true
            },
            phoneNumber: {
                value:'Value',
                isValid: true
            },
            address: {
                value: {},
                isValid: true
            },
            onChange: jest.fn()
        }

        const history = { push: jest.fn() }
		const wrapper = mount(<ContactDetails context={data} history={history} />)

        //Act
        wrapper.find('form').simulate('submit')


        //Assert
        expect(history.push).toHaveBeenCalledWith(getPageRoute(3))
    })

    describe('snapshot', () => {
		it('should render correctly',() => { 
			// Arrange
			const data = {
                emailAddress: {
                    value:'Value',
                    isValid: true
                },
                phoneNumber: {
                    value:'Value',
                    isValid: true
                },
                address: {
                    value: {},
                    isValid: true
                },
                onChange: jest.fn()
            }
            const history = {
                push: jest.fn()
            }
		
			// Act
			const tree = renderer
			.create(<ContactDetails context={data} history={history} />)
			.toJSON()
		
			// Assert
			expect(tree).toMatchSnapshot()
		})
	})
})
