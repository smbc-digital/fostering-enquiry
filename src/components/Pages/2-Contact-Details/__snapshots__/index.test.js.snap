// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetails snapshot should render correctly 1`] = `
Array [
  <h1 />,
  <h2>
    Your contact information
  </h2>,
  <p>
    We'll use this contact information to get in touch about your enquiry
  </p>,
  <form
    onSubmit={[Function]}
  >
    <label
      className=""
      htmlFor="emailAddress"
    >
      Email address
    </label>
    <input
      aria-label="Email address"
      aria-live="off"
      className=""
      id="emailAddress"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="emailAddress"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="email"
      value="Value"
    />
    <label
      className=""
      htmlFor="phoneNumber"
    >
      Phone number
    </label>
    <input
      aria-label="Phone number"
      aria-live="off"
      className=""
      id="phoneNumber"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="phoneNumber"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="tel"
      value="Value"
    />
    <h3
      aria-hidden="true"
      className={undefined}
    >
      Enter your address manually
    </h3>
    <label
      className=""
      htmlFor="addressLine1"
    >
      Address line 1
    </label>
    <input
      aria-label="Address line 1"
      aria-live="off"
      className=""
      id="addressLine1"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="addressLine1"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <h3
      aria-hidden="true"
      className="optional"
    >
      Address line 2
    </h3>
    <p
      className="optional"
    >
      (optional)
    </p>
    <label
      className="optional sr-only"
      htmlFor="addressLine2"
    >
      Address line 2 optional
    </label>
    <input
      aria-label=""
      aria-live=""
      className=""
      id="addressLine2"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="addressLine2"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <label
      className=""
      htmlFor="town"
    >
      Town
    </label>
    <input
      aria-label="Town"
      aria-live="off"
      className=""
      id="town"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="town"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <label
      className=""
      htmlFor="postcode"
    >
      Postcode
    </label>
    <input
      aria-label="Postcode"
      aria-live="off"
      className=""
      id="postcode"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="postcode"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <p>
      <a
        href="#"
        onClick={[Function]}
      >
         Use address finder 
      </a>
    </p>
    <button
      className="button-primary"
      disabled={false}
      id={undefined}
      onClick={undefined}
      type="submit"
    >
      Next step
    </button>
  </form>,
  <p>
    <a
      href="#"
      onClick={[Function]}
    >
      Previous
    </a>
  </p>,
]
`;
